### PR TITLE
docs(deployment): create Deploy with Docker page

### PR DESCRIPTION
**Pull request by hassankhan**
Story details: https://app.clubhouse.io/cube-dev/story/660

~Wait for https://github.com/cube-js/cube.js/pull/2892 to be merged first~

### CODE REVIEW

File: docs/content/Deployment/Platform-Docker.md
Comment (rpaik) [inline docs/content/Deployment/Platform-Docker.md:232]:
```suggestion
## Update to the latest version
```

Comment (rpaik) [reply docs/content/Deployment/Platform-Docker.md:232]: Not sure if this is better?

File: docs/content/Configuration/Extending-Cubejs.md
Comment (igorlukanin) [inline docs/content/Configuration/Extending-Cubejs.md:33]:
```suggestion
| container, make sure you mount the project _including_ the
```

File: docs/content/Deployment/Platform-Docker.md
Comment (igorlukanin) [inline docs/content/Deployment/Platform-Docker.md:13]: Having prerequisites is really cool. It also great to provide links to install or set up the prerequisites. (It also seems that there's no clear way to install "Docker CLI", just "[Docker Desktop](https://www.docker.com/products/docker-app)".)

File: docs/content/Deployment/Platform-Docker.md
Comment (igorlukanin) [inline docs/content/Deployment/Platform-Docker.md:31]: Is there any reason to not recommend running the latest version of Cube.js via the `latest` tag? (Putting `latest` here will also reduce the complexity of this config.)

File: docs/content/Deployment/Platform-Docker.md
Comment (igorlukanin) [inline docs/content/Deployment/Platform-Docker.md:18]: The example config is quite lengthy and contains a lot of things, so it might make sense to explain what's going to be run: a Cube.js API instance, a refresh worker, etc.

File: docs/content/Deployment/Platform-Docker.md
Comment (igorlukanin) [inline docs/content/Deployment/Platform-Docker.md:110]:
```suggestion
### Enable HTTPS/TLS
```

I think SSL is not a thing anymore :)

Comment (igorlukanin) [reply docs/content/Deployment/Platform-Docker.md:110]: Oh, I still see it in NGINX docs 😬

**Review by igorlukanin — APPROVED** [review]
Wonderful job, @hassankhan!

File: docs/content/Deployment/Platform-Docker.md
Comment (hassankhan) [reply docs/content/Deployment/Platform-Docker.md]: We didn't change it there to preserve references to the heading IIRC, but we should definitely update this one

File: docs/content/Deployment/Platform-Docker.md
Comment (hassankhan) [reply docs/content/Deployment/Platform-Docker.md:39]: In production, it's not recommended to use the `latest` tag but instead use a specific version number. Perhaps @ovr can confirm?

Comment (igorlukanin) [reply docs/content/Deployment/Platform-Docker.md:39]: Okay, makes sense!

Comment (ovr) [reply docs/content/Deployment/Platform-Docker.md:39]: @hassankhan I think we need to use a plugin for %VERSION, something like https://github.com/getsentry/sentry-docs/blob/master/src/gatsby/plugins/gatsby-remark-variables/index.js

**Review by keydunov — APPROVED** [review]
LGTM! I'd love to get @paveltiunov, @ovr, or @ilya-biryukov approval before the merge.

File: docs/content/Deployment/Platform-Docker.md
Comment (ovr) [inline docs/content/Deployment/Platform-Docker.md:123]: Maybe we need to point that Cube.js doesn't support HTTP2/3/Quick and it's better to use cube.js behind reverse proxy

File: docs/content/Deployment/Platform-Docker.md
Comment (ovr) [inline docs/content/Deployment/Platform-Docker.md:145]: I don't think that it's a good idea to write ssl_ciphers. There are different formats how it can be specified: like `ALL:!aNULL:!EXPORT56:RC4+RSA:+HIGH:+MEDIUM:+LOW:+SSLv2:+EXP` or `HIGH:!aNULL:!MD5` or by each one with `:` as separator.

Maybe it's better to write a comment above?

Comment (hassankhan) [reply docs/content/Deployment/Platform-Docker.md:145]: Can do, I just grabbed them from the LetsEncrypt NGINX example

File: docs/content/Deployment/Platform-Docker.md
Comment (hassankhan) [reply docs/content/Deployment/Platform-Docker.md:123]: Yeah good point, maybe I'll add a header specifically for reverse proxying

**Pull request merged**
